### PR TITLE
scale geometric error with tileset

### PR DESCRIPTION
**Pull request by Vineg**
#5330

### CODE REVIEW

**Comment by hpinkos** [comment]
Thanks @Vineg!  @lilleyse will review this soon

**Comment by Vineg** [comment]
@lilleyse this is ready for review

File: Source/Core/Matrix3.js [hunk 1]
```diff
@@ -1023,6 +1023,27 @@ define([
...
+
+    /**
+     * Factors rotation matrix from arbitrary Matrix3 by dividing components by scale
+     *
+     * @param {Matrix3} rotation-scale matrix
+     * @param {Matrix3} result rotation matrix with unit scale
+     * @returns {Matrix3} The modified result parameter.
+     */
+    Matrix3.getRotation = function(matrix, result) {
+        Matrix3.multiplyByScale(
```
Comment (hpinkos) [inline Source/Core/Matrix3.js]: Check that the arguments are defined:

```
        //>>includeStart('debug', pragmas.debug);
        Check.typeOf.object('matrix', matrix);
        Check.typeOf.object('result', result);
        //>>includeEnd('debug');
```

Comment (hpinkos) [reply Source/Core/Matrix3.js]: Also, this is better written as `result = Matrix3.multiplyByScale(.....`.  It makes the code more clear to someone not as familiar with Javascript who might not realize the result argument is being mutated

**Comment by hpinkos** [comment]
Can you add some unit tests for the new matrix function?

**Comment by hpinkos** [comment]
Those are all my comments. @lilleyse will do the final review.  Thanks @Vineg!

**Comment by Vineg** [comment]
@hpinkos @lilleyse done

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @Vineg! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by hpinkos** [comment]
@lilleyse can you review this?

**Comment by lilleyse** [comment]
Yes, sorry for the delay. I plan to review this soon.

**Comment by hpinkos** [comment]
@Vineg @geoscan-builder bump =)  let us know when these comments are addressed and this is ready for another look.  Thanks!

**Comment by Vineg** [comment]
Yeah, I hope to get back to this soon

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @Vineg! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @Vineg! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by Vineg** [comment]
@cesium-concierge stop

**Comment by Vineg** [comment]
@lilleyse applied proposed changes and moved getRotation to getMatrix3 as discussed in https://github.com/AnalyticalGraphicsInc/cesium/issues/7398

**Comment by hpinkos** [comment]
@lilleyse what's the status of this PR?

**Comment by lilleyse** [comment]
@hpinkos Waiting on my review which I'll finish up now.

File: Specs/Core/Matrix3Spec.js [hunk 2]
```diff
@@ -639,6 +639,41 @@ defineSuite([
...
+        result = Matrix3.getRotation(rotation, result);
+        expect(rotation).toEqualEpsilon(result, CesiumMath.EPSILON14);
+        expect(rotation).not.toBe(result);
+    });
+
+    it('getRotation throws without a matrix', function() {
+        expect(function() {
+            return Matrix3.getRotation();
+        }).toThrowDeveloperError();
+    });
```
Comment (lilleyse) [inline Specs/Core/Matrix3Spec.js]: Also add a separate check for `getRotation throws without a result`

**Comment by lilleyse** [comment]
Note to self after merging: this PR fixes both https://github.com/AnalyticalGraphicsInc/cesium/issues/5330 and https://github.com/AnalyticalGraphicsInc/cesium/issues/7398

**Comment by hpinkos** [comment]
I'm going to merge this into a branch and take over the rest of the clean up.  Thanks for all the work you did for this @Vineg!

**Pull request merged**